From ec8b264534fd76a1518eff1b7f3a28b861aaae2e Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 18 Oct 2018 20:12:38 +0300
Subject: [PATCH 684/767] staging: fsl-dpaa2/mac: read phy mode from device
 tree

If the a dpmac node defines its phy mode in the device tree using
the 'phy-mode' or the 'phy-connection-type' attributes this will take
precedence over the interface mode reported by the MC in the
dpmac attributes structure.

Signed-off-by: Ioana Ciornei <ioana.ciornei@nxp.com>
Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 9ea2dd674588..c55a06e3a50c 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -555,6 +555,13 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 		goto probe_fixed_link;
 	}
 
+	if_mode = of_get_phy_mode(dpmac_node);
+	if (if_mode >= 0) {
+		dev_dbg(dev, "\tusing if mode %s for eth_if %d\n",
+			phy_modes(if_mode), priv->attr.eth_if);
+		goto phy_connect;
+	}
+
 	if (priv->attr.eth_if < ARRAY_SIZE(dpaa2_mac_iface_mode)) {
 		if_mode = dpaa2_mac_iface_mode[priv->attr.eth_if];
 		dev_dbg(dev, "\tusing if mode %s for eth_if %d\n",
@@ -565,6 +572,7 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 		goto probe_fixed_link;
 	}
 
+phy_connect:
 	/* try to connect to the PHY */
 	netdev->phydev = of_phy_connect(netdev, phy_node,
 					&dpaa2_mac_link_changed, 0, if_mode);
-- 
2.17.0

